src/Raven.Server/Documents/Sharding/Subscriptions/SubscriptionConnectionsStateOrchestrator.cs

```diff
@@ -142,15 +143,17 @@ public override async Task HandleConnectionExceptionAsync(OrchestratedSubscripti
         {
             DropSubscription(new SubscriptionClosedException("Got unexpected exception, dropping workers connections.", canReconnect: true, e));
         }
+
+        await t;
     }
 
     public override void Dispose()
     {
-        DisposeWorkers();
+        DisposeWorkers(waitForSubscriptionTask: false);
```

Comment:
this could hold up the `DatabaseContext.Dispose()` for up to 60 sec

Reply:
why?

Reply:
I meant if we wait for subscription task for finish here, it can take up to 60 sec, this dispose called only on ServerStore dispose or delete of subscription task